src/main/java/ch/njol/skript/expressions/ExprFurnaceTime.java

```diff
@@ -0,0 +1,246 @@
+			case DELETE -> changeFurnaces(event, furnace -> furnace.setCookTime((short) 0));
+			case REMOVE -> changeFurnaces(event, furnace -> furnace.setCookTime((short) Math2.fit(0, furnace.getCookTime() - providedTime, Integer.MAX_VALUE)));
+			case ADD -> changeFurnaces(event, furnace -> furnace.setCookTime((short) Math2.fit(0, furnace.getCookTime() + providedTime, Integer.MAX_VALUE)));
+		}
+	}
+
+	private void changeTotalCookTime(Event event, int providedTime, ChangeMode mode) {
+		switch (mode) {
+			case SET -> {
+				if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+					startEvent.setTotalCookTime(providedTime);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setCookTimeTotal(providedTime));
+				}
+			}
+			case DELETE -> {
+				if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+					startEvent.setTotalCookTime(0);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setCookTimeTotal(0));
+				}
+			}
+			case REMOVE -> {
+				if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+					startEvent.setTotalCookTime(Math.min(startEvent.getTotalCookTime() - providedTime, 0));
+				} else {
+					changeFurnaces(event, furnace -> furnace.setCookTimeTotal(Math2.fit(0, furnace.getCookTimeTotal() - providedTime, Integer.MAX_VALUE)));
+				}
+			}
+			case ADD -> {
+				if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+					startEvent.setTotalCookTime(startEvent.getTotalCookTime() + providedTime);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setCookTimeTotal(Math2.fit(0, furnace.getCookTimeTotal() + providedTime, Integer.MAX_VALUE)));
+				}
+			}
+		}
+	}
+
+	private void changeBurnTime(Event event, int providedTime, ChangeMode mode) {
+		switch (mode) {
+			case SET -> {
+				if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+					burnEvent.setBurnTime(providedTime);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setBurnTime((short) providedTime));
+				}
+			}
+			case DELETE -> {
+				if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+					burnEvent.setBurnTime(0);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setBurnTime((short) 0));
+				}
+			}
+			case REMOVE -> {
+				if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+					burnEvent.setBurnTime(Math.min(burnEvent.getBurnTime() - providedTime, 0));
+				} else {
+					changeFurnaces(event, furnace -> furnace.setBurnTime((short) Math2.fit(0, furnace.getBurnTime() - providedTime, Integer.MAX_VALUE)));
+				}
+			}
+			case ADD -> {
+				if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+					burnEvent.setBurnTime(burnEvent.getBurnTime() + providedTime);
+				} else {
+					changeFurnaces(event, furnace -> furnace.setBurnTime((short) Math2.fit(0, furnace.getBurnTime() + providedTime, Integer.MAX_VALUE)));
+				}
+			}
+		}
+	}
```

Comment:
great candidate here for a helper method to handle all the explicitly block stuff